### PR TITLE
Initial steps towards 20x Return to Pasadena

### PR DESCRIPTION
**Pull request by owendelong**
## Description of PR
Update files to build configurations for PCC (SCaLE 20x)

## Previous Behavior
Built configurations for Hilton (SCaLE 19x)

## New Behavior
Buids configurations for PCC (SCaLE 20x)

## Tests
Not thoroughly. Ran configuration scripts and did some spot checking of some generated configuration files.

### CODE REVIEW

File: switch-configuration/config/scripts/bulk_local_load_switches [hunk 1]
```diff
@@ -76,6 +74,7 @@ while (1) {
     my $arp = `arp -n $IP | grep $IP`;
     chomp($arp);
     $arp = lc($arp);
+    # Harry regex to support differing arp outputs (BSD vs. Linux vs. MacOS)
```
Comment (sarcasticadmin) [inline switch-configuration/config/scripts/bulk_local_load_switches:77]: thanks for clarifying

File: switch-configuration/config/scripts/update_switches [hunk 2]
```diff
@@ -3,8 +3,11 @@
 # Push an updated copy of the configuration file to a switch and activate it.
 #
 
-# Not yet ready for use... Do not use
-die "Don't use this script yet. It's not ready"
+# Deprecated in favor of override_switches
+# Code kept around as template to later improve communications in override_switches and bulk_load_local_switches
+
+# Deprecated... Do not use
+die "Don't use this script yet. It's deprecated. See override_switches or bulk_load_ocal_switches."
```
Comment (sarcasticadmin) [inline switch-configuration/config/scripts/update_switches:10]: thanks for pointing us in the right direction with what to use instead :+1:

File: switch-configuration/config/switchtypes [hunk 3]
```diff
@@ -9,61 +9,62 @@
...
+ExpoC5		44	103	2001:470:f0fb:103::200:44	Booth		W.9		Loud		ex4200-48p	00:26:88:7c:41:7f
+ExpoAW		45	103	2001:470:f0fb:103::200:45	Booth		W.9		Normal		ex4200-48p	00:26:88:6e:b1:ff
+CTF5		46	504	2001:470:f0fb:504::200:46	cfCTF		I.9		Loud		ex4200-48p	00:26:88:7f:0e:7f
+CTF6		47	504	2001:470:f0fb:504::200:47	cfCTF		I.9		Normal		ex4200-48p	00:26:88:6e:9f:ff
+SpareA		48	103	2001:470:f0fb:103::200:48	cfRoom		Z.9		??		ex4200-48p	84:18:88:ac:d7:7f
+SpareB		49	103	2001:470:f0fb:103::200:49	cfRoom		Z.9		??		ex4200-48p	84:18:88:ab:df:ff
+SpareC		50	103	2001:470:f0fb:103::200:50	cfRoom		Z.9		??		ex4200-48p	84:18:88:ad:4e:ff
+SpareD		51	103	2001:470:f0fb:103::200:51	cfRoom		Z.9		??		ex4200-24t	02:00:00:00:00:51
+SpareE		52	103	2001:470:f0fb:103::200:52	cfRoom		Z.9		??		ex4200-24t	02:00:00:00:00:52
+SpareF		53	103	2001:470:f0fb:103::200:53	cfRoom		Z.9		??		ex4200-24t	2c:6b:f5:8e:5c:ff
```
Comment (sarcasticadmin) [inline switch-configuration/config/switchtypes:70]: vlan `103` vs `503` seems to line up like we would expect :+1:

File: switch-configuration/config/vlans.d/Expo [hunk 4]
```diff
@@ -10,7 +10,8 @@ VLAN	exSigns				107	2001:470:f0fb:107::/64	0.0.0.0/0	Signs network (Expo Center)
 VLAN	exStaff				108	2001:470:f0fb:108::/64	10.0.8.0/24	Staff Wireless Network
 //109 not used
 VLAN	exRegistration  	110	2001:470:f0fb:110::/64	10.0.10.0/24	Registration Network (Expo Center) IPv6 Only
-VLAN	exAkamai			111	2001:470:f0fb:111::/64	0.0.0.0/0	Special public Akamai Cache Cluster network (has IPv4 from convention center)
+// Akamai cache is deprecated
```
Comment (sarcasticadmin) [inline switch-configuration/config/vlans.d/Expo:13]: Good catch, yep akamai is definitely deprecated

File: switch-configuration/todo [hunk 5]
```diff
@@ -7,13 +7,12 @@ x   3.	L3 Interfaces for management VLAN on switches
 x   4.	Scripts to build configuration files
 x       Switches
 x       Routers
-    5.	Scripts for switch installation and maintenance (Robert)
-            Started, but likely replace with ansible
+    5.	Scripts for switch installation and maintenance (Owen)
```
Comment (sarcasticadmin) [inline switch-configuration/todo:10]: man thats old

File: switch-configuration/config/scripts/bulk_local_load_switches [hunk 6]
```diff
@@ -12,7 +12,7 @@
 #
 #	The system where this software is running should have an interface set up on that same network.
 #
-#	1. Delete 192.168.255.76 from the local ARP cache.
+#	1. Delete 192.168.255.76 from the local ARP cache. -- Deprecated
```
Comment (sarcasticadmin) [inline switch-configuration/config/scripts/bulk_local_load_switches:15]: is this really no longer a problem with the arp cache?

File: switch-configuration/config/scripts/README.md [hunk 7]
```diff
@@ -7,10 +7,16 @@ Part of [SCaLE's](https://www.socallinuxexpo.org/) on-site expo network configur
...
+
+* generate_ps_refs.pl
+-- This script takes the EPS files generated by the build_switch_configs.pl script and generates a file containing pages of switch reference maps intended for viewing on a laptop. It can be viewed on a phone, but is not optimized or likely to provide a particularly good user experience in a phone environment.
 
 * switch_template.pl
--- This script should not be directly executed. It is a collection of library functions to support build_switch_config.pl. Most of the actual work occurs inside these library functions.
+-- This script should not be directly executed. It is a collection of library functions to support other scripts here. Most of the actual work occurs inside these library functions, especially reading and interpreting the various configuration files and generating actual configuration date for the switches.
```
Comment (sarcasticadmin) [inline switch-configuration/config/scripts/README.md:19]: thanks for updating the docs

**Review by sarcasticadmin — APPROVED** [review]
@owendelong looks like a great step toward getting the switch configs back to where they need to be for 20x and pasadena

:shipit:

File: switch-configuration/todo (hunk 5 above)
Comment (owendelong) [reply switch-configuration/todo:10]: True

File: switch-configuration/config/scripts/bulk_local_load_switches (hunk 6 above)
Comment (owendelong) [reply switch-configuration/config/scripts/bulk_local_load_switches:15]: Turns out yes. The arp cache clears when it detects link state drop.

OTOH, deleting from the ARP cache created a sudo dependency that wasn't worth the hassle.

**Pull request merged**